### PR TITLE
Add ci step to validate incorrect replacement fork

### PR DESCRIPTION
**Pull request by MonzElmasry**
#### Proposed Changes ####

Add CI step to check for proper vendoring 

#### Types of Changes ####

CI
#### Verification ####

CI on pull request should have an extra step `validate_go_mods` that errors out when vendoring from a personal fork
#### Linked Issues ####
https://github.com/k3s-io/k3s/issues/2720

### CODE REVIEW

File: scripts/test-mods [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -e -x
+
+res=$(go mod edit --json | jq '.Replace[] | select(.Old.Path | contains("k8s.io/")) | .New.Path' | grep -v k3s-io | wc -l)
+if [ $res -gt 0 ];then
+  echo "Incorrect kubernetes replacement fork in go.mod"
+  exit 1
+else
+    exit 0
```
Comment (brandond) [inline scripts/test-mods:9]: Inconsistent indentation

File: Dockerfile.test.mod.dapper [hunk 2]
```diff
@@ -0,0 +1,12 @@
+ARG GOLANG=golang:1.16.2-alpine3.12
+FROM ${GOLANG}
+
+RUN apk -U --no-cache add bash jq
+ENV DAPPER_SOURCE /go/src/github.com/rancher/k3s/
+ENV DAPPER_DOCKER_SOCKET true
```
Comment (brandond) [inline Dockerfile.test.mod.dapper:6]: Does not appear to need Docker for anything

**Review by brandond — APPROVED** [review]
Couple nits, other than that lgtm!

**Review by brandond — APPROVED** [review]

File: scripts/test-mods [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -e -x
+
+res=$(go mod edit --json | jq '.Replace[] | select(.Old.Path | contains("k8s.io/")) | .New.Path' | grep -v k3s-io | wc -l)
+if [ $res -gt 0 ];then
+  echo "Incorrect kubernetes replacement fork in go.mod"
+  exit 1
+else
+  exit 0
+fi
```
Comment (briandowns) [inline scripts/test-mods:10]:
```suggestion
fi

```

**Review by briandowns — APPROVED** [review]

**Pull request merged**